src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -359,25 +387,29 @@ public void Remove(object key)
                 return;
             }
 
-            objectsTable.Remove(key);
-            objectsArray.RemoveAt(index);
+            EnsureObjectsTable();
+            EnsureObjectsArray();
+            _objectsTable.Remove(key);
+            _objectsArray.RemoveAt(index);
         }
 
 #region IDictionary implementation
         public virtual IDictionaryEnumerator GetEnumerator()
         {
-            return new OrderedDictionaryEnumerator(objectsArray, OrderedDictionaryEnumerator.DictionaryEntry);
+            EnsureObjectsArray();
+            return new OrderedDictionaryEnumerator(_objectsArray, OrderedDictionaryEnumerator.DictionaryEntry);
         }
 #endregion
 
 #region IEnumerable implementation
         IEnumerator IEnumerable.GetEnumerator()
         {
-            return new OrderedDictionaryEnumerator(objectsArray, OrderedDictionaryEnumerator.DictionaryEntry);
+            EnsureObjectsArray();
+            return new OrderedDictionaryEnumerator(_objectsArray, OrderedDictionaryEnumerator.DictionaryEntry);
         }
 #endregion
 
-#region ISerializable implementation 
+#region ISerializable implementation
```

Comment:
We need `EnsureObjectsArray()` added to `GetObjectData` as it reads `_objectsArray`.
This is causing some of the tests to fail in CI.

Reply:
https://mc.dot.net/#/user/iSazonov/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/551ba9e70c3e5ace47c8ccb36a0815cfc074fb3a/workItem/System.Runtime.Serialization.Formatters.Tests/analysis/xunit/System.Runtime.Serialization.Formatters.Tests.BinaryFormatterTests~2FRoundtripManyObjectsInOneStream

Reply:
Fixed.

Reply:
I can not open your link - I see only progress bar :-(